src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotUseInsecureDeserializerJsonNetWithoutBinderTests.cs

```diff
@@ -674,32 +674,38 @@ T Method<T>(JsonReader jr)
         [Fact]
         public async Task TypeNameHandlingNoneBinderNonNull_JsonSerializer_Populate_NoDiagnostic()
         {
-            await VerifyCSharpWithJsonNetAsync(@"
+#if NETCOREAPP
+            var serializationNamespace = "";
+#else
+            var serializationNamespace = "using System.Runtime.Serialization;";
+#endif
```

Comment:
What problems did you run into with System.Runtime.Serialization.SerializationBinder on .NET Core?

Reply:
Okay, so I think Newtonsoft.Json 10.0.1 is weird. Should change AdditionalMetadataReferences.DefaultWithNewtonsoftJson to use version 12.0.3.

Then the test can be:
```csharp
        [Fact]
        public async Task TypeNameHandlingNoneBinderNonNull_JsonSerializer_Populate_NoDiagnostic()
        {
            await VerifyCSharpWithJsonNetAsync(@"
using System;
using Newtonsoft.Json;

class Blah
{
    private Func<System.Runtime.Serialization.SerializationBinder> SbGetter;

    object Method(JsonReader jr)
    {
        System.Runtime.Serialization.SerializationBinder sb = SbGetter();
        if (sb != null)
        {
            JsonSerializer serializer = new JsonSerializer();
            serializer.Binder = sb;
            object o = new object();
            serializer.Populate(jr, o);
            return o;
        }
        else
        {
            return null;
        }
    }
}");
        }
```

Reply:
Should we still have a test for 10.x or is that no longer supported by the analyzers?

Reply:
The tests have been updated to test against both 10.0.1 and 12.0.1.

Reply:
I'd just remove the 10.x reference. Not that it isn't supported; just has a funky interface in its .NET Standard library. Totally fine with keeping it too.